### PR TITLE
Unfiltered professions search page (1812)

### PR DESCRIPTION
**Pull request by bryangaledxw**
This PR adds an unfiltered professions search page, following Ming's prototype - https://rpr-prototype.herokuapp.com/beta-sprint-2/public-facing-v1/selected-service

### CODE REVIEW

File: cypress/integration/professions/search/search.spec.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+describe('Searching a profession', () => {
+  it('I can view an unfiltered list of profession', () => {
+    cy.visit('/professions/search');
+
+    cy.get('body').should('contain', 'Registered Trademark Attorney');
```
Comment (pezholio) [inline cypress/integration/professions/search/search.spec.ts:5]: Can we use the translations here?

File: src/professions/search/search.controller.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    private readonly professionsService: ProfessionsService,
+    private readonly industriesService: IndustriesService,
+    private readonly i18nService: I18nService,
+  ) {}
+
+  @Get()
+  @Render('professions/search/search')
+  async search(@Req() request: Request): Promise<SearchTemplate> {
+    return {
+      ...(await this.creatSearchResults(new FilterDto())),
```
Comment (pezholio) [inline src/professions/search/search.controller.ts:23]: Typo 👍

File: src/professions/search/search.controller.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    @Body() filter: FilterDto,
+    @Req() request: Request,
+  ): Promise<SearchTemplate> {
+    return {
+      ...(await this.creatSearchResults(filter)),
+      backLink: backLink(request),
+    };
+  }
+
+  private async creatSearchResults(
```
Comment (pezholio) [inline src/professions/search/search.controller.ts:40]: Could we break this out into a `SearchService`? I think I'd prefer the various different variables that get returned at the end to be broken out into their own methods too, to make it more easier to test and to follow. It'll then make the controller tests simpler and seperate out the concerns a bit.

Comment (pezholio) [reply src/professions/search/search.controller.ts:40]: That said, I know you're working on the next bit now, so I'm happy for this to go in as is and we can iterate 👍

File: views/professions/search/search.njk [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+      
+      <p class="govuk-body-m"><span class="govuk-!-font-size-36 govuk-!-font-weight-bold">{{ professions.length }}</span> {{ "profession" if (professions.count === 1) else "professions" }} found.</p>
+
+      <hr class="govuk-section-break govuk-section-break--l govuk-section-break--visible">
+
+      <div class="rpr-listing__profession-container">
+
+        {% for profession in professions %}
+          <h2>
+            <a class="govuk-link govuk-heading-l rpr-listing__profession-title" href="profession-1-plastic-surgery">{{profession.name}}</a>
```
Comment (pezholio) [inline views/professions/search/search.njk:116]:
```suggestion
<a class="govuk-link govuk-heading-l rpr-listing__profession-title" href="/professions/{{ profession.id }}">{{profession.name}}</a>
```

File: src/professions/search/search.controller.ts [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+@Controller('professions/search')
+export class SearchController {
+  constructor(
+    private readonly professionsService: ProfessionsService,
+    private readonly industriesService: IndustriesService,
+    private readonly i18nService: I18nService,
+  ) {}
+
+  @Get()
+  @Render('professions/search/search')
```
Comment (pezholio) [inline src/professions/search/search.controller.ts:20]: I think `index` is a better name for this template

File: src/professions/search/search.controller.ts [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+  async search(@Req() request: Request): Promise<SearchTemplate> {
```
Comment (pezholio) [inline src/professions/search/search.controller.ts:21]: Again, as this is the root action `index` is the better name here

File: src/professions/search/search.controller.ts [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+    return {
+      ...(await this.creatSearchResults(new FilterDto())),
+      backLink: backLink(request),
+    };
+  }
+
+  @Post()
+  @Render('professions/search/search')
+  async searchPost(
```
Comment (pezholio) [inline src/professions/search/search.controller.ts:30]: This can probably be called `create`, as we're `create`ing a search query.

**Review by pezholio — APPROVED** [review]
Just a couple of typos and naming considerations. One thing to think about, but I know you're working on the next thing, so happy for this to go in as is and iterate on the next card 👍

File: cypress/integration/professions/search/search.spec.ts (hunk 1 above)
Comment (bryangaledxw) [reply cypress/integration/professions/search/search.spec.ts:5]: I don't think so? - this is the name in the DB and the idea of localised names for professions isn't really something we've tackled

Comment (pezholio) [reply cypress/integration/professions/search/search.spec.ts:5]: Oh, yeah, you're right. Carry onj 😄

File: src/professions/search/search.controller.ts (hunk 3 above)
Comment (bryangaledxw) [reply src/professions/search/search.controller.ts]: Yep, this is the first think in the upcoming PR

File: views/professions/search/search.njk (hunk 4 above)
Comment (bryangaledxw) [reply views/professions/search/search.njk]: Woops!

**Pull request merged**